gui/src-tauri/src/main.rs

```diff
@@ -38,22 +39,53 @@ fn update_window_state(
 	Ok(())
 }
 
-fn main() {
+#[tauri::command]
+fn logging(msg: String) {
+	log::info!(target: "webview", "[webview] {}", msg)
+}
+
+#[tauri::command]
+fn erroring(msg: String) {
+	log::error!(target: "webview", "[webview] {}", msg)
+}
+
+#[tauri::command]
+fn warning(msg: String) {
+	log::warn!(target: "webview", "[webview] {}", msg)
+}
+
+fn main() -> Result<()> {
+	log_panics::init();
+	let hook = panic::take_hook();
 	// Make an error dialog box when panicking
-	panic::set_hook(Box::new(|panic_info| {
-		println!("{}", panic_info);
+	panic::set_hook(Box::new(move |panic_info| {
+		hook(panic_info);
 		show_error(&panic_info.to_string());
 	}));
 
 	let cli = Cli::parse();
+	let tauri_context = tauri::generate_context!();
 
 	// Set up loggers and global handlers
-	{
-		if std::env::var_os("RUST_LOG").is_none() {
-			std::env::set_var("RUST_LOG", "info")
-		}
-		pretty_env_logger::init();
-	}
+	let _logger = {
```

Comment:
no need to assign to an unused var.

Reply:
it needs to be alive, its buffering logs and writing them to a file, more info in their [docs.rs](https://docs.rs/flexi_logger/0.25.5/flexi_logger/code_examples/index.html#choose-the-write-mode)